### PR TITLE
Move x86 Linux setup to prereqs script

### PR DESCRIPTION
**Pull request by jonsimantov**
Move x86 Linux setup to prereqs - this lets us add a retry to the Linux package installation, which can sometimes fail due to network connectivity.

(The setup is still present in the build script, but won't do anything if it determines that the required packages are installed.)
***

Desktop checks and integration tests in this PR.
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [x] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: scripts/gha/utils.py [hunk 1]
```diff
@@ -218,3 +218,45 @@ def clean_vcpkg_temp_data():
...
+    if gha_build:
+      # Workaround for GitHub runners, which have an incompatibility between the
+      # 64-bit and 32-bit versions of the Ubuntu package libpcre2-8-0. Downgrade
+      # the installed 64-bit version of the library to get around this issue.
+      # This will presumably be fixed in a future Ubuntu update. (If you remove
+      # it, remove the workaround further down this function as well.)
+      packages = ['--allow-downgrades'] + packages + ['libpcre2-8-0=10.34-7']
+
+    # First check if these packages exist on the machine already
+    devnull = open(os.devnull, "w")
```
Comment (dconeybe) [inline scripts/gha/utils.py:244]: drive-by comment: use a `with` block to ensure `devnull` gets closed even if `subprocess.run()` raises an exception.

i.e.

```
with open(os.devnull, "w") as devnull:
  subprocess.run(...)
```

Comment (dconeybe) [inline scripts/gha/utils.py:244]: Another totally optional drive-by comment: Consider writing to a temp file. IMO it's a bit more idiomatic in Python.

i.e.

```
with tempfile.TemporaryFile() as stdout_tempfile:
  subprocess.run(..., stdout=stdout_tempfile)
```

File: scripts/gha/install_prereqs_desktop.py [hunk 2]
```diff
@@ -29,62 +29,76 @@
...
+    if not utils.is_command_installed('go'):
+      if utils.is_linux_os():
+          # sudo apt install -y golang
+          utils.run_command(['apt', 'install', '-y','golang'], as_root=True)
+      elif utils.is_mac_os():
+          # brew install protobuf
+          utils.run_command(['brew', 'install', 'go'])
+
+    # Install openssl on linux/mac if its not installed already
+    if not utils.is_command_installed('go'):
```
Comment (dconeybe) [inline scripts/gha/install_prereqs_desktop.py:58]: Is it time to finally fix this "if" statement? It's checking for the "go" binary to determine if "openssl" is installed, which looks like a copy/paste error from the "if" block above.

**Review by a-maurice — APPROVED** [review]

File: scripts/gha/utils.py (hunk 1 above)
Comment (jonsimantov) [reply scripts/gha/utils.py:244]: Thanks for the suggestion - trying not to change any of the code since this is just copied from build_desktop, but will consider for future changes.

File: scripts/gha/install_prereqs_desktop.py (hunk 2 above)
Comment (jonsimantov) [reply scripts/gha/install_prereqs_desktop.py:58]: Yes, it is.

**Review by a-maurice — APPROVED** [review]
